### DISCUSSION TITLE
Unify component and component_block

### DISCUSSION THREAD

**Discussion by gone** [django-components/django-components#120] [Ideas]
I haven't thought too hard about the complexity here, but from a pure ergonomics pov it doesn't make sense to have two tags.

**Comment by EmilStenstrom** [comment]
@gone I agree. Not sure it's possible though. 

The core issue is: How does django differentiate between `{% component %}Some content` and `{% component %}Some content{% endcomponent %}`? 

Ways around this:
* Parsing the whole document to the end, and keep track of when a component tag starts, when a nested tag starts, and when a tag ends. When we know the number of close tags, and what level they are on, and what start tag the correspond to, we can parse again. I think this might slow things down too much.
* We can add a parameter to component that changes how the tag works. But is this really better than two tags?
* Maybe skip the simple version, and only supply the slotted one? Not sure I think this would be an improvement.

Other ideas?

**Comment by JuroOravec** [comment]
Closing as this was added in v0.5
